tools/populate_dandiset_asset_yamls.py

```diff
@@ -39,43 +39,52 @@ def main():
             relpath = f.relative_to(dspath)
             print("Processing", f)
             sha256_digest = get_digest(f)
-            errors = validate(f)
-            if errors:
-                print("PYNWB ERRORS:")
-                for e in errors:
-                    print(f" - {e}")
-                with pynwb_errs.open("a") as fp:
-                    print(relpath, file=fp)
+            if f.suffix == ".nwb":
+                errors = validate(f)
+                if errors:
+                    print("PYNWB ERRORS:")
                     for e in errors:
-                        print(f" - {e}", file=fp)
-                    print(file=fp)
-            try:
-                metadata = nwb2asset(
-                    f, digest=sha256_digest, digest_type="SHA256"
-                ).json_dict()
-            except Exception as e:
-                print(f"CONVERSION ERROR: {e}")
-                with conversion_errs.open("a") as fp:
-                    print(relpath, file=fp)
-                    traceback.print_exc(file=fp)
-                    print(file=fp)
+                        print(f" - {e}")
+                    with pynwb_errs.open("a") as fp:
+                        print(relpath, file=fp)
+                        for e in errors:
+                            print(f" - {e}", file=fp)
+                        print(file=fp)
+                try:
+                    metadata = nwb2asset(
+                        f, digest=sha256_digest, digest_type="SHA256"
+                    ).json_dict()
+                except Exception as e:
+                    print(f"CONVERSION ERROR: {e}")
+                    with conversion_errs.open("a") as fp:
+                        print(relpath, file=fp)
+                        traceback.print_exc(file=fp)
+                        print(file=fp)
+                    metadata = {
```

Comment:
may be just assign it ahead of the `if` and thus avoid bottom `else`/duplication?

Reply:
Done.